### ISSUE TITLE
Disable Style/YodaCondition by default?

### ISSUE THREAD

**Issue by Nondv** [rubocop-hq/rubocop#6030]
Hi!

I couldnt find a real reason behind forbidding yoda conditions (#4145). What is wrong with them? Unnatural?

Well, yes. But thing is they can improve drastically readability in some cases. Especially in long-ass expressions:

```ruby
return true if app_users.with_confirmed_email.average_age > 18
# vs
return true if 18 < app_users.with_confirmed_email.average_age
# or something like
users = app_users.with_confirmed_email
return true if users.average_age  > 18
# or moving to User class method
```

or with long names:

```ruby
return some_value if some_object.some_long_method_name > 18
return some_value if 18 < some_object.some_long_method_name
```

Though in most cases it just should be moved into a predicate method it is not really necessary *all the time*.

Yoda conditions are not even covered by style guide. So I suggest disabling this cop by default or writing a good explanation why you should not use yoda conditions at all.

**Comment by bbatsov** [comment]
> Unnatural?

That seems to be the common consensus. 

> Well, yes. But thing is they can improve drastically readability in some cases. Especially in long-ass expressions:

I guess that's extremely subjective, as I don't see any readability improvements. :-) For me the only real benefit yoda conditions have is that they are NPE-safe. So far you're the first person to bring this subject up, which leads me to believe the default setting is fine for most people. 

> Yoda conditions are not even covered by style guide. So I suggest disabling this cop by default or writing a good explanation why you should not use yoda conditions at all.

An oversight on our part that we should address.

**Comment by Drenmi** [comment]
> For me the only real benefit yoda conditions have is that they are NPE-safe.

For me this is a downside, because it tends to mask real problems. 😅
